CRM/Event/BAO/Query.php

```diff
@@ -253,6 +253,10 @@ public static function where(&$query) {
    * @param $query
    */
   public static function whereClauseSingle(&$values, &$query) {
+    $checkPermission = TRUE;
+    if (!empty($query->_skipPermission)) {
+      $checkPermission = FALSE;
+    }
```

Comment:
This could be written in 1 line: 

    $checkPermission = empty($query->_skipPermission);